sphinx_needs/directives/need.py

```diff
@@ -460,82 +438,56 @@ def format_need_nodes(
-                        if report_dead_links:
-                            message = f"Need '{need['id']}' has unknown outgoing link '{need_id_full}' in field '{link_type['option']}'"
-                            # if the need has been imported from an external URL,
-                            # we want to provide that URL as the location of the warning,
-                            # otherwise we use the location of the need in the source file
-                            if need["is_external"]:
-                                log_warning(
-                                    LOGGER,
-                                    f"{need['external_url']}: {message}",
-                                    "external_link_outgoing",
-                                    None,
-                                )
-                            else:
-                                log_warning(
-                                    LOGGER,
-                                    message,
-                                    "link_outgoing",
-                                    location=(need["docname"], need["lineno"]),
-                                )
-
-
-def create_back_links(needs: NeedsMutable, config: NeedsSphinxConfig) -> None:
-    """Create back-links in all found needs.
-
-    These are fields for each link type, ``<link_name>_back``,
-    which contain a list of all IDs of needs that link to the current need.
-    """
-    for links in config.extra_links:
-        option = links["option"]
-        option_back = f"{option}_back"
+    for key, need in needs.items():
+        dead_links = []
 
-        for key, need in needs.items():
-            need_link_value: list[str] = (
-                [need[option]] if isinstance(need[option], str) else need[option]
-            )
-            for need_id_full in need_link_value:
+        for link_type, references in need.iter_links_items():
+            for need_id_full in references:
                 need_id_main, need_id_part = split_need_id(need_id_full)
-
-                if need_id_main in needs:
-                    if key not in needs[need_id_main][option_back]:
-                        needs[need_id_main][option_back].append(key)
-
-                    # Handling of links to need_parts inside a need
-                    if need_id_part and need_id_part in needs[need_id_main]["parts"]:
-                        if (
-                            option_back
-                            not in needs[need_id_main]["parts"][need_id_part]
-                        ):
-                            needs[need_id_main]["parts"][need_id_part][option_back] = []  # type: ignore[literal-required]
-                        needs[need_id_main]["parts"][need_id_part][option_back].append(  # type: ignore[literal-required]
-                            key
-                        )
+                if linked_need := needs.get(need_id_main):
+                    linked_need.add_backlink(link_type, key)
```

Comment:
Note, this is where it always creates a backlink to the parent, as well as the part (as per the current code)
This could easily be changed, if we feel it is correct to do so